EIPS/eip-712.md

```diff
@@ -404,49 +384,14 @@ The Solidity expression `keccak256(someInstance)` for an instance `someInstance`
-[ex-js]: https://github.com/ethereum/EIPs/blob/master/assets/eip-712/Example.js
-
-## Implementation
-
-To be done before this EIP can be considered accepted:
-
-*   [x] Finalize specification of structure hashing
-*   [x] Domain separators
-*   [x] Add test vectors
-*   [x] Review specification
-
-To be done before this EIP can be considered "Final":
-
-*   [x] Implement `eth_signTypedData` in major RPC providers.
-*   [x] Implement `web3.eth.signTypedData` in Web3 providers.
-*   [x] Implement `keccak256` struct hashing in Solidity.
-
-**Rationale for marking goals as done:**
-* `eth_signTypedData`
+[ex-sol]: ../assets/eip-712/Example.sol
+[ex-js]: ../assets/eip-712/Example.js
 
-MetaMask has had a series of `eth_signTypedData` since 2017 ([article](https://medium.com/metamask/scaling-web3-with-signtypeddata-91d6efc8b290)), and is an accepted standard in the wallet ecosystem.
-
-* `web3.eth.signTypedData`
-
-While this api does **not** exist in the web3.js library as of this writing, it is included as an experimental feature in the Ethers library [here](https://medium.com/metamask/scaling-web3-with-signtypeddata-91d6efc8b290). It can also be called from the Web3.js library by using the `sendAsync` method:
-
-```javascript
-web3.currentProvider.sendAsync({
-  method: 'eth_signTypedData',
-  params: [msgParams, from]
-```
-In addition, the `signTypedData` function is of limited utility in web3 convenience libraries, as it is generally relevant to the interface that stores the private keys used for signing, and with the excepetionof burner wallets, this is almost always done in practice within a wallet.
-
-* `keccak256` struct hashing
-
-While the `keccak256` function in Solidity cannot be invoked directly on structs, the introduction of `abi.encode` gives a sufficient and well-accepted avenue to hashing structs.
-
-* review of specification
-
-In the time since EIP-712 has been proposed, it has enjoyed wide ecosystem adoption, and has become the foundation for a number of other EIPs building on top of it. (For example, ERC20 token Permits (transaction-less token approvals) in [EIP-2612](./eip-2612.md).) MetaMask uses EIP-712 for more readable output to present users, and other wallets have followed suit. The widespread traction and use of this EIP over the course of years justify regarding as sufficiently reviewed.
+## Security Considerations
 
+### Note on replay attacks
 
+This standard is only about signing messages and verifying signatures. In many practical applications, signed messages are used to authorize an action, for example an exchange of tokens. It is _very important_ that implementers make sure the application behaves correctly when it sees the same signed message twice. For example, the repeated message should be rejected or the authorized action should be idempotent. How this is implemented is specific to the application and out of scope for this standard.
```

Comment:
Since you mentioned the replay attack, it's probably also worthy of mentioning that in many case once a message is signed, they are no guarantee their distribution will be restricted or they delivery will be ensured and confirmed, therefore the applications need to consider such cases to ensure safety.

Reply:
What do you mean by "restricted distribution"?

Reply:
> What do you mean by "restricted distribution"?

I imagine he means that the signature should be assumed to possibly become public knowledge.

Reply:
@Pandapip1 is right. Yes, the signer shall expect the signature MAY AND MAY NOT be public knowledge.